roles/confluent.kafka_broker/tasks/set_principal.yml

```diff
@@ -19,7 +19,35 @@
     kafka_broker_principal: "User:{{ kerberos_kafka_broker_primary }}"
   when: listener['sasl_protocol'] | default(sasl_protocol) | normalize_sasl_protocol == 'GSSAPI'
 
-- name: Extract Common Name from Keystore - SSL Mutual Auth
+- name: Extract Common Names from Keystore - Provided SSL Mapping Rule
+  # Search lines with Entry type: "PrivateKeyEntry" and return that line and all after, ca cert is of type "trustedCertEntry"
+  # Search for all lines with "Owner" key
+  # Extract DNAME from line
+  # Remove spaces after commas
+  shell: |
+    keytool -list -keystore {{kafka_broker_keystore_path}} \
+        -storepass {{kafka_broker_keystore_storepass}} -v \
+        | grep PrivateKeyEntry -A1000 \
+        | grep Owner \
+        | cut -d ":" -f2 \
+        | cut -c2- \
+        | sed 's/\s*,\s*/,/g'
+  register: distinguished_names_from_keystore
+  changed_when: false
+  when:
+    - listener['sasl_protocol'] | default(sasl_protocol) | normalize_sasl_protocol == 'none'
+    - listener['ssl_enabled'] | default(ssl_enabled) | bool
+    - listener['ssl_mutual_auth_enabled'] | default(ssl_mutual_auth_enabled) | bool
+    - kafka_broker_final_properties['ssl.principal.mapping.rules'] is defined
+  no_log: "{{ mask_sensitive_logs|bool }}"
+
+- name: Set the distinguished name - Provided SSL Mapping Rule
+  set_fact:
+    distinguished_name: "{{distinguished_names_from_keystore.stdout|resolve_principal(kafka_broker_final_properties['ssl.principal.mapping.rules'])}}"
+  when:
+    - distinguished_names_from_keystore.stdout is defined | default('undefined') | bool
+
+- name: Extract Common Name from Keystore
```

Comment:
Is this task and `Extract Common Names from Keystore - Provided SSL Mapping Rule` exactly the same? 
If so you should be able to have one task, and then conditionally do the next ones

Reply:
There is slight difference in both tasks. The new one extracts all the common names whereas the legacy one extracts only the first common name. 
`| grep Owner \` vs `| grep Owner  -m1\`

Reply:
Those are not exactly same. The legacy one extracts the first like with `Onwer` keyword, where as the new implementation takes all values and tries to map the rules

Reply:
hmm and why have the difference? Im pretty sure in a keystore there can be a certificate chain and the very first cert is the one that use used in mtls connections

Reply:
I wasn't sure the very first cert is the one always used in mtls connection. I have updated the code to reuse in both cases now.

Reply:
```suggestion
- name: Extract Common Name from Keystore - SSL Mutual Auth
```